Update release note with gRPC, Kafka, RabbitMQ, NATS updates (#2100)
ballerina-platform/ballerina-release: release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md

##### `sql` Package
- Added schema client abstraction to support metadata retrieval from SQL databases. The implementation for the connectors will be added soon

##### `grpc` Package

> - Introduced message level annotations for the proto descriptor instead of a centralized proto descriptor

Reviewer: ```suggestion
- Introduced message-level annotations for the proto descriptor instead of a centralized proto descriptor
```

##### `random` Package
- Updated the `createDecimal()` function to be cryptographically secure

##### `grpc` Package
- Added sample client calls with dummy values to generated client files
- Removed caller client object when generating code in client mode

> - Changed the ServiceDescriptor annotation to Descriptor (Please update the service annotation if stub files are regenerated for the existing gRPC services)

Reviewer 2: ```suggestion
- Added a new `grpc:Descriptor` annotation for services as a replacement for the current `grpc:ServiceDescriptor`. Both annotations are supported now to maintain backward compatibility. The ServiceDescriptor will be removed in the future. (Please update the service annotation if stub files are regenerated for the existing gRPC services)
```

> - Added a new `grpc:Descriptor` annotation for services as a replacement for the current `grpc:ServiceDescriptor`. Both annotations are supported now to maintain backward compatibility. The ServiceDescriptor will be removed in the future. (Please update the service annotation if stub files are regenerated for the existing gRPC services)

Reviewer 2: ```suggestion
- Added a new `grpc:Descriptor` annotation for services as a replacement for the current `grpc:ServiceDescriptor`. Both annotations are supported now to maintain backward compatibility. The `grpc:ServiceDescriptor` will be removed in the future. (Please update the service annotation if stub files are regenerated for the existing gRPC services)
```
